### PR TITLE
[TK-07459] Proxy Metrics

### PR DESCRIPTION
**Pull request by neonphog**
proxy-cli will now print out the following info from the proxy server:

```json
{
  "sub_transport": {
    "url": "kitsune-quic://10.0.0.103:51755",
    "connection_count": 1
  },
  "url": "kitsune-proxy://ytP_jyLtax5epvW8uJ58MOFTkY5jS4BO8tpNutbHImA/kitsune-quic/h/10.0.0.103/p/51755/--",
  "proxy_count": 0,
  "tokio_task_count": 11,
  "sys_info": {
    "used_mem_kb": 9264,
    "proc_cpu_usage_pct_1000": 0,
    "tx_bytes_per_sec": 440,
    "rx_bytes_per_sec": 1257
  }
}
```

### CODE REVIEW

File: crates/kitsune_p2p/types/src/metrics.rs [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+static RX_BYTES_PER_SEC: AtomicU64 = AtomicU64::new(0);
+
+/// Spawns a tokio task with given future/async block.
+/// Captures a new TaskCounter instance to track task count.
+#[macro_export]
+macro_rules! metric_task {
+    ($task:expr) => {{
+        let counter = $crate::metrics::MetricTaskCounter::new();
+        let fut = { $task };
+        ::tokio::task::spawn(async move {
```
Comment (neonphog) [inline crates/kitsune_p2p/types/src/metrics.rs:24]:
```suggestion
        $crate::dependencies::tokio::task::spawn(async move {
```

we don't know that they directly imported tokio...

**Review by freesig — APPROVED** [review]
Overall looks good, just a little worried about wrapping large blocks in macros. I know that this is a problem with tokio select! but might be fine with your macro

File: crates/kitsune_p2p/proxy/src/bin/proxy-cli.rs [hunk 2]
```diff
@@ -43,7 +44,7 @@ async fn inner() -> TransportResult<()> {
     let (listener, mut events) =
         spawn_kitsune_proxy_listener(proxy_config, listener, events).await?;
 
-    tokio::task::spawn(async move {
+    metric_task!(async move {
```
[the remark this answers is not in the document]
Comment (neonphog) [reply crates/kitsune_p2p/proxy/src/bin/proxy-cli.rs:47]: Good call. Pretty sure I can switch this to a plain function - will work on that this morning.

[the remark this answers is not in the document]
Comment (neonphog) [reply crates/kitsune_p2p/proxy/src/bin/proxy-cli.rs]: Done.

**Pull request merged**